### PR TITLE
Added tags

### LINKED ISSUE #4: Tag
Abstract class to add tags to product and people, for example, a product can be a pizza, spicy, and only sold on mondays... And a person can be in 3rd year, part of the DI, and a cafeteria employee (so 3 tags on the Product and 3 tas on the person here).

### LINKED ISSUE #5: ProductTag
Tag to differentiate the products, by having for exemple a pizza tag, or a soda tag

### LINKED ISSUE #10: PersonTag
Tag to represent informations on someone, like if he works at the cafetaria or not, what promotion is the person in

### LINKED ISSUE #11: YearTag
Tag to represent what year is the person in, like 3rd year, or PEIP

### PR DESCRIPTION
**Pull request by Banalian**
closes #4
closes #5
closes #10
closes #11

### CODE REVIEW

File: src/main/java/fr/ppptours/cafetaria/model/tags/Tag.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package fr.ppptours.cafetaria.model.tags;
+
+/**
+ * Abstract class for all tags
+ */
```
Comment (Nowp) [inline src/main/java/fr/ppptours/cafetaria/model/tags/Tag.java:5]:
```suggestion
/**
 * Base class for tags.
 */
```

File: src/main/java/fr/ppptours/cafetaria/model/tags/Tag.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+     */
+    private String tagName;
+
+    public Tag(String tagName) {
+        this.tagName = tagName;
+    }
+
+    /**
+     * @return The tagName
+     */
```
Comment (Nowp) [inline src/main/java/fr/ppptours/cafetaria/model/tags/Tag.java:19]:
```suggestion
```

**Review by Nowp — CHANGES_REQUESTED** [review]
Documentation could be improved, no need to specify when things are obvious or even redundant

File: src/main/java/fr/ppptours/cafetaria/model/tags/YearTag.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package fr.ppptours.cafetaria.model.tags;
+
+/**
+ * Tag Class to represent a year/class ( for example a client being in 3rd year, or in PEIP)
```
Comment (Schizopinguin) [inline src/main/java/fr/ppptours/cafetaria/model/tags/YearTag.java:4]: "or in PEIP", I agree but an external user wont understand this statement

Comment (Banalian) [reply src/main/java/fr/ppptours/cafetaria/model/tags/YearTag.java]: I changed it to "preparatory class" to be more user-friendly

File: src/main/java/fr/ppptours/cafetaria/model/tags/Tag.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    public Tag(String tagName) {
+        this.tagName = tagName;
+    }
+
+    public String getTagName() {
+        return tagName;
+    }
+
+    /**
+     * Set a new tag Name
```
Comment (Nowp) [inline src/main/java/fr/ppptours/cafetaria/model/tags/Tag.java:22]:
```suggestion
     * Sets a new tag Name
```

**Review by Nowp — APPROVED** [review]

**Review by Schizopinguin — APPROVED** [review]
good job

**Pull request merged**